File: Projects/GLTetris/src/grid.c
/*
 * This file is part of GLTetris
 * 
 * GLTetris GLUT/OpenGL based Tetris.
 * Copyright (C) 2008  Frederic-Gerald Morcos
 * 
 * GLTetris is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * GLTetris is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with GLTetris.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdlib.h>
#include <GL/glut.h>

#include "grid.h"
#include "global.h"

/* create a new grid */
grid *grid_new ()
{
	grid *newg = malloc (sizeof (grid));
	newg->step = GRID_STEP;
	newg->width = GRID_COLS;
	newg->height = GRID_ROWS;
	newg->r = G_R;
	newg->g = G_G;
	newg->b = G_B;
	
	return newg;
}

/* draw the grid, should be called from the 
 * display callback function
 */
void grid_draw (grid *g)
{
	int i = 0;
	
	glColor3f (g->r, g->g, g->b);
	
	glBegin (GL_LINES);
	while (i <= GRID_COLS * GRID_STEP)
	{
		glVertex2i (i, 0);
		glVertex2i (i, GRID_ROWS * GRID_STEP);
		
		i += GRID_STEP;
	}
	
	i = 0;
	
	while (i <= GRID_ROWS * GRID_STEP)
	{
		glVertex2i (0, i);
		glVertex2i (GRID_COLS * GRID_STEP, i);
		i += GRID_STEP;
	}
	glEnd ();
}

File: Projects/OpenGrafik/opengrafik_repo_20090321/trunk/Grid.m
/*	
 *	This file is part of OpenGrafik.
 *
 *	Copyright 2009 Fred Morcos <fred.morcos@gmail.com>
 *
 *	OpenGrafik is free software: you can redistribute it and/or modify
 *	it under the terms of the GNU General Public License as published by
 *	the Free Software Foundation, either version 3 of the License, or
 *	(at your option) any later version.
 *
 *	OpenGrafik is distributed in the hope that it will be useful,
 *	but WITHOUT ANY WARRANTY; without even the implied warranty of
 *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *	GNU General Public License for more details.
 *
 *	You should have received a copy of the GNU General Public License
 *	along with OpenGrafik.  If not, see <http://www.gnu.org/licenses/>.
 */

#import "Grid.h"

@implementation Grid

- init {
	self = [super init];
	cellSize = 20;
	return self;
}

- setCellSize: (int) size {
	cellSize = size;
	[self changedProperty: @"CellSize"];
	return self;
}

- (int) cellSize {
	return cellSize;
}

- render {
	if (visible == YES) {
		for (int i = cellSize; i < rect.size.width; i += cellSize) {
			[path moveToPoint: NSMakePoint(i, 0)];
			[path lineToPoint: NSMakePoint(i, rect.size.height)];
		}
		
		for (int i = cellSize; i < rect.size.height; i += cellSize) {
			[path moveToPoint: NSMakePoint(0, i)];
			[path lineToPoint: NSMakePoint(rect.size.width, i)];
		}
	}
	return self;
}

- stroke {
	[NSBezierPath clipRect: rect];
	[bgColor set];
	[NSBezierPath fillRect: rect];
	[fgColor set];
	[path stroke];
	return self;
}

@end

